Sources/SwiftHttp/HttpHeaderKey.swift

```diff
@@ -8,21 +8,294 @@
+	public static let summary: HttpHeaderKey = "Summary"
+	public static let supersedes: HttpHeaderKey = "Supersedes"
+	public static let tLSReportDomain: HttpHeaderKey = "TLS-Report-Domain"
+	public static let tLSReportSubmitter: HttpHeaderKey = "TLS-Report-Submitter"
+	public static let tLSRequired: HttpHeaderKey = "TLS-Required"
+	public static let to: HttpHeaderKey = "To"
+	public static let userAgent: HttpHeaderKey = "User-Agent"
+	public static let vBRInfo: HttpHeaderKey = "VBR-Info"
+	public static let x400ContentIdentifier: HttpHeaderKey = "X400-Content-Identifier"
+	public static let x400ContentReturn: HttpHeaderKey = "X400-Content-Return"
+	public static let x400ContentType: HttpHeaderKey = "X400-Content-Type"
+	public static let x400MTSIdentifier: HttpHeaderKey = "X400-MTS-Identifier"
+	public static let x400Originator: HttpHeaderKey = "X400-Originator"
+	public static let x400Received: HttpHeaderKey = "X400-Received"
+	public static let x400Recipients: HttpHeaderKey = "X400-Recipients"
+	public static let x400Trace: HttpHeaderKey = "X400-Trace"
+	public static let xrefcase: HttpHeaderKey = "Xrefcase"
+	/// Provisional Message Header Field Names
+	public static let apparentlyTo: HttpHeaderKey = "Apparently-To"
+	public static let author: HttpHeaderKey = "Author"
+	public static let eDIINTFeatures: HttpHeaderKey = "EDIINT-Features"
+	public static let eesstVersion: HttpHeaderKey = "Eesst-Version"
+	public static let errorsTo: HttpHeaderKey = "Errors-To"
+	public static let formSub: HttpHeaderKey = "Form-Sub"
+	public static let jabberID: HttpHeaderKey = "Jabber-ID"
+	public static let mMHSAuthorizingUsers: HttpHeaderKey = "MMHS-Authorizing-Users"
+	public static let privicon: HttpHeaderKey = "Privicon"
+	public static let sIOLabel: HttpHeaderKey = "SIO-Label"
+	public static let sIOLabelHistory: HttpHeaderKey = "SIO-Label-History"
+	public static let xArchivedAt: HttpHeaderKey = "X-Archived-At"
+	public static let xMittente: HttpHeaderKey = "X-Mittente"
+	public static let xPGPSig: HttpHeaderKey = "X-PGP-Sig"
+	public static let xRicevuta: HttpHeaderKey = "X-Ricevuta"
+	public static let xRiferimentoMessageID: HttpHeaderKey = "X-Riferimento-Message-ID"
+	public static let xTipoRicevuta: HttpHeaderKey = "X-TipoRicevuta"
+	public static let xTrasporto: HttpHeaderKey = "X-Trasporto"
+	public static let xVerificaSicurezza: HttpHeaderKey = "X-VerificaSicurezza"
+}
 
-    /// Type safe key using a `HttpHeader` value
-    case key(HttpHeader)
-    
-    /// Custom header key as a `String` value
-    case custom(String)
-    
-    /// Raw String value of the header key
-    public var rawValue: String {
-        switch self {
-        case .key(let header):
-            return header.rawValue
-        case .custom(let value):
-            return value
-        }
-    }
+extension HttpHeaderKey: Hashable, Codable, ExpressibleByStringLiteral, StringProtocol, RawRepresentable, CustomStringConvertible {
```

Comment:
What are these extensions for? When do you need these? 🤔

Reply:
Also could you provide some one-liner docs for these methods please?

Reply:
Maybe some test cases could be helpful too... :)

Reply:
HttpHeaderKey is a fundamental type, it's just a string wrapper, so it's useful to implement some base protocols
RawRepresentable I implemented in order to backward compatibility with enum from previous versions

Reply:
ExpressibleByStringLiteral is very convenience

Reply:
I added some comments

Reply:
All these methods are just wrappers, I don't think we need tests here,

Reply:
I removed `StringProtocol`